webhook/stats_reporter.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhook
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+	admissionv1beta1 "k8s.io/api/admission/v1beta1"
+	"knative.dev/pkg/metrics"
+)
+
+const (
+	requestCountName     = "request_count"
+	requestLatenciesName = "request_latencies"
+)
+
+var (
+	requestCountM = stats.Int64(
+		requestCountName,
+		"The number of requests that are routed to webhook",
+		stats.UnitDimensionless)
+	responseTimeInMsecM = stats.Float64(
+		"request_latencies",
+		"The response time in milliseconds",
+		stats.UnitMilliseconds)
+
+	defaultLatencyDistribution = view.Distribution(5, 10, 20, 40, 60, 80, 100, 150, 200, 250, 300, 350, 400, 450, 500, 600, 700, 800, 900, 1000, 2000, 5000, 10000, 20000, 50000, 100000)
```

Comment:
I think @grantr created a library for building distributions with a nice bucketing. Not sure where it's living right now, though.

Reply:
Mmmh, I don't see it anywhere in this repo. I've reached out to @grantr offline. In the meanwhile, I'll stick with this (which matches the activator's latency distribution).

Reply:
Grant got back to me, I've copied the bucketing function into the pkg/metrics package so that other components can use it in the future.

Reply:
@evankanderson actually wrote the code in https://github.com/knative/eventing/pull/1177#discussion_r292208873, I just copied it :smile: